### PR TITLE
sink(ticdc): use a global variable to ensure that the batch ID is monotonically increasing

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/8132

### What is changed and how it works?

use a global variable to ensure that the batch ID is monotonically increasing. See more in the code comments.
#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/processor/sinkmanager/table_sink_worker.go [hunk 1]
```diff
@@ -109,7 +131,9 @@ func validateAndAdjustBound(changefeedID model.ChangeFeedID,
 }
 
 func (w *sinkWorker) handleTask(ctx context.Context, task *sinkTask) (finalErr error) {
-	advancer := newTableSinkAdvancer(task, w.splitTxn, w.sinkMemQuota, requestMemSize)
+	// We need to use a new batch ID for each task.
+	batchID := batchID.Add(1)
+	advancer := newTableSinkAdvancer(task, w.splitTxn, w.sinkMemQuota, requestMemSize, batchID)
```
Comment (hicqu) [inline cdc/processor/sinkmanager/table_sink_worker.go:136]: Why not access global `batchID` in Advancer directly?

Comment (hi-rustin) [reply cdc/processor/sinkmanager/table_sink_worker.go:136]: It breaks a lot of code and is not easy to test and maintain.

Comment (hi-rustin) [reply cdc/processor/sinkmanager/table_sink_worker.go:136]: Also, it reduces a lot of atomic operations.

**Review by amyangfei — APPROVED** [review]

**Pull request merged**